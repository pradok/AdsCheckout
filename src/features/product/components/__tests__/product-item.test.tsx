import * as React from 'react';
import {shallow} from 'enzyme';

import ProductItem, {Props} from '../product-item';

const productItem: Props = {
  product: {
    price: 269.99,
    title: 'Classic Ad',
  },
  onClickAddToCart: jest.fn(),
};

it('<ProductItem> renders correctly', () => {
  const component = shallow(
    <ProductItem {...productItem} />
  );
  expect(component).toMatchSnapshot();
});

it('<ProductItem> Add to cart button click', () => {
  const component = shallow(
    <ProductItem {...productItem} />
  );
  component.find('.add-cart-btn').simulate('click');
  expect(productItem.onClickAddToCart).toHaveBeenCalled();
});
