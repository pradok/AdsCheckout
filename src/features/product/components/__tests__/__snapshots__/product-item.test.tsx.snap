// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductItem> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductItem
    onClickAddToCart={[MockFunction]}
    product={
      Object {
        "price": 269.99,
        "title": "Classic Ad",
      }
    }
    promo="Get 3 for 2 deal on Classic Ads"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <h3>
          Classic Ad
        </h3>,
        <h4>
          $
          269.99
        </h4>,
        <ul>
          <li>
            Get 3 for 2 deal on Classic Ads
          </li>
        </ul>,
        <styled.button
          className="add-cart-btn"
          onClick={[MockFunction]}
        >
          Add to cart
        </styled.button>,
      ],
      "className": "product-item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Classic Ad",
        },
        "ref": null,
        "rendered": "Classic Ad",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "$",
            269.99,
          ],
        },
        "ref": null,
        "rendered": Array [
          "$",
          269.99,
        ],
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <li>
            Get 3 for 2 deal on Classic Ads
          </li>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Get 3 for 2 deal on Classic Ads",
          },
          "ref": null,
          "rendered": "Get 3 for 2 deal on Classic Ads",
          "type": "li",
        },
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Add to cart",
          "className": "add-cart-btn",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "Add to cart",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <h3>
            Classic Ad
          </h3>,
          <h4>
            $
            269.99
          </h4>,
          <ul>
            <li>
              Get 3 for 2 deal on Classic Ads
            </li>
          </ul>,
          <styled.button
            className="add-cart-btn"
            onClick={[MockFunction]}
          >
            Add to cart
          </styled.button>,
        ],
        "className": "product-item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Classic Ad",
          },
          "ref": null,
          "rendered": "Classic Ad",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "$",
              269.99,
            ],
          },
          "ref": null,
          "rendered": Array [
            "$",
            269.99,
          ],
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <li>
              Get 3 for 2 deal on Classic Ads
            </li>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Get 3 for 2 deal on Classic Ads",
            },
            "ref": null,
            "rendered": "Get 3 for 2 deal on Classic Ads",
            "type": "li",
          },
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Add to cart",
            "className": "add-cart-btn",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "Add to cart",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
